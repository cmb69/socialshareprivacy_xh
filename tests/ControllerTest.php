<?php

namespace Socialshareprivacy;

use ApprovalTests\Approvals;
use PHPUnit\Framework\TestCase;
use Plib\FakeRequest;
use Plib\View;

class ControllerTest extends TestCase
{
    /** @var array<string,string> */
    private array $conf;
    private View $view;

    protected function setUp(): void
    {
        global $plugin_cf, $plugin_tx;
        $plugin_cf = XH_includeVar("./config/config.php", "plugin_cf");
        $plugin_tx = XH_includeVar("./languages/en.php", "plugin_tx");
        $this->conf = $plugin_cf["socialshareprivacy"];
        $this->view = new View("./views/", $plugin_tx["socialshareprivacy"]);
    }

    private function sut(): Controller
    {
        return new Controller("./", $this->conf, $this->view);
    }

    public function testRendersWidget(): void
    {
        $request = new FakeRequest();
        $response = $this->sut()->init($request);
        Approvals::verifyHtml($response->output());
    }

    public function testUsesConfiguredUrl(): void
    {
        $url = "https://cmsimple-xh.org/";
        $this->conf["url"] = $url;
        $request = new FakeRequest();
        $response = $this->sut()->init($request);
        $this->assertStringContainsString(urlencode($url), $response->output());
    }
}
