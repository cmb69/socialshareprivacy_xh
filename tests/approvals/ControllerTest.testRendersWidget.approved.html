
<script type="module" src="./socialshareprivacy.js"></script>
<figure class="socialshareprivacy_share">
  <figcaption>Share this page</figcaption>
  <div class="socialshareprivacy_links">
    <a class="socialshareprivacy_facebook" href="https://www.facebook.com/sharer/sharer.php?u=http%3A%2F%2Fexample.com%2F" data-share-url="https://www.facebook.com/sharer/sharer.php?u=" target="_blank">
      <svg role="img" focusable="false" enable-background="new 0 0 56.693 56.693" height="2em" id="Layer_1" version="1.1" viewBox="0 0 56.693 56.693" width="2em" xml:space="preserve" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"><title>
        Share on facebook</title>
        <desc>facebook logo</desc>
        <path d="M40.43,21.739h-7.645v-5.014c0-1.883,1.248-2.322,2.127-2.322c0.877,0,5.395,0,5.395,0V6.125l-7.43-0.029  c-8.248,0-10.125,6.174-10.125,10.125v5.518h-4.77v8.53h4.77c0,10.947,0,24.137,0,24.137h10.033c0,0,0-13.32,0-24.137h6.77  L40.43,21.739z" fill="#ffffff"/>
      </svg>
    </a>
    <a class="socialshareprivacy_x" href="https://x.com/intent/post?url=http%3A%2F%2Fexample.com%2F" data-share-url="https://x.com/intent/post?url=" target="_blank">
      <svg role="img" focusable="false" xmlns="http://www.w3.org/2000/svg" width="1.5em" height="1.5em" viewBox="0 0 1200 1227" fill="none" style="padding: 0 0 0 0.2em;">
        <title>Share on X</title>
        <desc>X logo</desc>
        <g clip-path="url(#clip0_1_2)"><path d="M714.163 519.284L1160.89 0H1055.03L667.137 450.887L357.328 0H0L468.492 681.821L0 1226.37H105.866L515.491 750.218L842.672 1226.37H1200L714.137 519.284H714.163ZM569.165 687.828L521.697 619.934L144.011 79.6944H306.615L611.412 515.685L658.88 583.579L1055.08 1150.3H892.476L569.165 687.854V687.828Z" fill="white"/></g><defs><clipPath id="clip0_1_2"><rect width="1200" height="1227" fill="white"/></clipPath></defs>
      </svg>
    </a>
    <a class="socialshareprivacy_xing" href="https://www.xing.com/social/share/spi?url=http%3A%2F%2Fexample.com%2F" data-share-url="https://www.xing.com/social/share/spi?url=" target="_blank">
      <svg role="img" focusable="false" width="2em" height="2em" enable-background="new 0 0 195.3381 216" id="xing" version="1.1" viewBox="0 0 195.3381 216" xml:space="preserve" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
        <title>Share on XING</title>
        <desc>XING logo</desc>
        <g><polygon fill="#C1DB66" points="195.3381,0 148.6252,0 70.8516,137 68.832,140.5584 70.9181,144 114.5548,216 162.2304,216    117.1212,137.7812  "/><polygon fill="#63B473" points="73.8284,98.2892 52.3108,48 9.9316,48 31.4512,98.2892 0,156 42.3792,156 74.8244,100.6172  "/></g>
      </svg>
    </a>
    <a class="socialshareprivacy_linkedin" href="https://www.linkedin.com/sharing/share-offsite/?url=http%3A%2F%2Fexample.com%2F" data-share-url="https://www.linkedin.com/sharing/share-offsite/?url=" target="_blank">
      <svg enable-background="new 0 0 32 32" height="1.5em" id="Layer_1" version="1.0" viewBox="0 0 32 32" width="1.5em" xml:space="preserve" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" style="padding: 0 0 0.1em 0.2em;">
        <title>Share on LinkedIn</title>
        <desc>LinkedIn logo</desc>
        <g><rect fill="#ffffff" height="23" width="7" y="9"/><path d="M24.003,9C20,9,18.89,10.312,18,12V9h-7v23h7V19c0-2,0-4,3.5-4s3.5,2,3.5,4v13h7V19C32,13,31,9,24.003,9z" fill="#ffffff"/><circle cx="3.5" cy="3.5" fill="#ffffff" r="3.5"/></g><g/><g/><g/><g/><g/><g/>
      </svg>
    </a>
  </div>
</figure>
